system/smp.c

```diff
@@ -548,13 +824,35 @@ void smp_init(bool smp_enable)
 
     if (smp_enable) {
         (void)(find_cpus_in_madt() || find_cpus_in_floating_mp_struct());
-
     }
 
     for (int i = 0; i < num_available_cpus; i++) {
         apic_id_to_cpu_num[cpu_num_to_apic_id[i]] = i;
     }
 
+    if (smp_enable) {
+        if (!find_numa_nodes_in_srat()) {
+            // Do nothing.
+        }
+    }
+
+    for (int i = 0; i < num_available_cpus; i++) {
+        uint32_t proximity_domain_idx = apic_id_to_proximity_domain_idx[i];
+        cpus_in_proximity_domain[proximity_domain_idx]++;
+    }
+
+    // Create dummy entries when there's no (proper) memory affinity information.
+    if (num_memory_affinity_ranges == 0 || num_proximity_domains == 0) {
+        extern uintptr_t pm_limit_lower;
```

Comment:
Not convinced this is preferable to having a reference to `enable_numa` as you are still referencing variables in app!

I would have just set `num_proximity domains` to 0 and checked for that in `main()` after calling `smp_init()`.

Reply:
Heh, right. Missed that extern reference when writing the cover message. Will fix.

Reply:
Fixed by removing that block and setting `enable_numa` to `false` in `main()` below the call to `smp_init()`, which is itself below the call to `config_init()`.